### PR TITLE
CLI command

### LINKED ISSUE #8: Cli
- flags:
   - name - load balancer name / default random name
   - capacity - max number of nodes / default 10
   - whitelist - comma separated node ids, if submitted only whitelisted nodes allowed
   - fee -percentage
   - selection - round robin or random

### PR DESCRIPTION
**Pull request by MakMuftic**
### Changes

Expanded `start` CLI command with all defined flags 

- `auth-secret`
- `name`
- `capacity`
- `whitelist`
- `fee`
- `selection`

Extracted all setup and server starting logic from the `start` method.

### Issues

closes #8 

### Example

@mpetrun5 @mpetrunic Do you think these **[OPTIONAL]** and **[REQUIRED]** prefixes are needed?

image

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -1,55 +1,85 @@
...
+	startCmd.Flags().StringVar(
+		&name,
+		"name",
+		fmt.Sprintf("load-balancer-%s", random.String(12, random.Alphabetic)),
+		"[OPTIONAL] public name for load balancer, autogenerated name used if omitted")
+
+	startCmd.Flags().Int64Var(
+		&capacity,
+		"capacity",
+		10,
```
Comment (mpetrunic) [inline cmd/start.go]: probably should be unlimited by default

File: cmd/start.go [hunk 2]
```diff
@@ -1,55 +1,85 @@
...
+	startCmd.Flags().StringArrayVar(
+		&whitelist,
+		"whitelist",
+		nil,
+		"[OPTIONAL] comma separated list of node id-s, if provided only these nodes will be allowed to connect")
+
+	startCmd.Flags().Float32Var(
+		&fee,
+		"fee",
+		0,
```
Comment (mpetrunic) [inline cmd/start.go]: we can make 10% by default so it's not required

File: cmd/start.go [hunk 3]
```diff
@@ -1,55 +1,85 @@
...
-	err = database.Close()
-	log.Fatal(err)
+	loadbalancer.StartLoadBalancerServer(loadbalancer.Properties{
+		AuthSecret: authSecret,
+		Name:       name,
+		Capacity:   capacity,
+		Whitelist:  whitelist,
+		Fee:        fee,
+		Selection:  selection,
+	}, "4000")
```
Comment (mpetrunic) [inline cmd/start.go]: port should also be cli flag

File: internal/loadbalancer/server.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+type Properties struct {
+	AuthSecret string
+	Name string
+	Capacity int64
+	Whitelist []string
+	Fee float32
+	Selection string
+}
+
+func StartLoadBalancerServer(props Properties, port string) {
```
Comment (mpetrunic) [inline internal/loadbalancer/server.go]: port should be number?

File: internal/loadbalancer/server.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+	// init database
+	database, err := storm.Open("vedran-load-balancer.db")
+	if err != nil {
+		// terminate app: unable to start database connection
+		log.Fatal(fmt.Sprintf("Unable to start vedran load balancer: %v", err))
+	}
+
+	// start server
+	log.Println("Starting vedran load balancer on port :4000...")
```
Comment (mpetrunic) [inline internal/loadbalancer/server.go]: use real port value instead of hardoded 4000

File: cmd/start.go (hunk 2 above)
Comment (mpetrunic) [reply cmd/start.go]: also validate fee is between 0 and 100

Comment (mpetrunic) [reply cmd/start.go]: or 0 -1 since you are using float32

File: cmd/start.go (hunk 1 above)
Comment (mpetrunic) [reply cmd/start.go]: ensure it's higher than -1, we can use -1 as unlimited

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by mpetrunic — DISMISSED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**